src/api/app/models/obs_factory/distribution_strategy_factory.rb

```diff
@@ -114,8 +112,8 @@ def totest_version
     # @return [String] version string
     def published_version
       begin
-        f = open(repo_url)
-      rescue ::OpenURI::HTTPError
+        f = URI.parse(repo_url)
```

Comment:
you're telling me URI.parse opens the URL?

Reply:
[Yes](https://ruby-doc.org/stdlib-2.5.0/libdoc/uri/rdoc/URI.html#method-c-parse), well one of it's subclasses.
So this should work for FTP, HTTP and HTTPS (and some other protocols we don't care about).

Reply:
And while I am looking through the docu... we could even write this as URI(repo_url).

Reply:
this is creating an URI object not an IO stream

Reply:
the 2nd blog post you linked does URI(url).open

Reply:
> coolo 19 minutes ago  Member
> this is creating an URI object not an IO stream

That's not saying much, does it? Given that ruby has inheritence:-) And even without inheriting from IO a class could always have it's own `read` method defined.

I just did a brief look into the [URI code](https://github.com/ruby/ruby/tree/trunk/lib/uri) and didn't find were it comes from (URI inherits from Object). In case you want to dive into it.

```ruby
irb(main):005:0> URI::HTTPS.instance_methods
=> [:buffer_open, :request_uri, :read, ...
irb(main):002:0> URI("http://download.opensuse.org/tumbleweed/repo/oss/media.1/media").read
=> "openSUSE - openSUSE-20180726-i586-x86_64-Build393.2-Media\nopenSUSE-20180726-i586-x86_64-Build393.2\n1\n"
```

Reply:
Hmm, you must have a different irb 
```
NoMethodError (undefined method `read' for #<URI::HTTP:0x000000000249e2e0>)

Reply:
ah, you need to require open-uri and it will patch URI::HTTPS

Reply:
Fun fact, [open-uri](https://github.com/ruby/ruby/blob/trunk/lib/open-uri.rb#L45) overwrites URI#open, when loaded.

Reply:
so better not remove the require?

Reply:
> @coolo an hour ago
> Hmm, you must have a different irb
>
> NoMethodError (undefined method `read' for #<URI::HTTP:0x000000000249e2e0>)

That's what I get when adding pry binding to the test as well. I guess it depends on whether some module get's loaded or not.
Which is why I changed the code now to using Net::HTTP now.